table/table_test.go

```diff
@@ -128,3 +138,235 @@ func (t *TableTestSuite) TestSnapshotByName() {
+
+	files := make([]string, 0)
+	for i := range 5 {
+		filePath := fmt.Sprintf("%s/unpartitioned_file_not_found/test-%d.parquet", t.location, i)
+		t.writeParquet(tbl.FS().(iceio.WriteFileIO), filePath, t.arrTbl)
+		files = append(files, filePath)
+	}
+
+	files = append(files, t.location+"/unpartitioned_file_not_found/unknown.parquet")
+	tx := tbl.NewTransaction()
+	err := tx.AddFiles(files, nil, false)
+	t.Error(err)
+	t.ErrorContains(err, "no such file or directory")
+}
+
+func (t *TableWritingTestSuite) TestAddFilesUnpartitionedHasFieldIDs() {
+	ident := table.Identifier{"default", "unpartitioned_table_with_ids_v" + strconv.Itoa(t.formatVersion)}
+	tbl := t.createTable(ident, t.formatVersion,
+		*iceberg.UnpartitionedSpec, t.tableSchema)
+
+	t.NotNil(tbl)
+
+	files := make([]string, 0)
+	for i := range 5 {
+		filePath := fmt.Sprintf("%s/unpartitioned_with_ids/test-%d.parquet", t.location, i)
+		t.writeParquet(tbl.FS().(iceio.WriteFileIO), filePath, t.arrTblWithIDs)
+		files = append(files, filePath)
+	}
+
+	tx := tbl.NewTransaction()
+	err := tx.AddFiles(files, nil, false)
+	t.Error(err)
+	t.ErrorIs(err, iceberg.ErrNotImplemented)
+}
+
+func (t *TableWritingTestSuite) TestAddFilesFailsSchemaMismatch() {
+	ident := table.Identifier{"default", "unpartitioned_table_schema_mismatch_v" + strconv.Itoa(t.formatVersion)}
+	tbl := t.createTable(ident, t.formatVersion,
+		*iceberg.UnpartitionedSpec, t.tableSchema)
+	t.Require().NotNil(tbl)
+
+	wrongSchema := arrow.NewSchema([]arrow.Field{
+		{Name: "foo", Type: arrow.FixedWidthTypes.Boolean, Nullable: true},
+		{Name: "bar", Type: arrow.BinaryTypes.String, Nullable: true},
+		{Name: "baz", Type: arrow.BinaryTypes.String, Nullable: true}, // should be int32
+		{Name: "qux", Type: arrow.PrimitiveTypes.Date32, Nullable: true},
+	}, nil)
+
+	filePath := fmt.Sprintf("%s/unpartitioned_schema_mismatch_v%d/test.parquet", t.location, t.formatVersion)
+	mismatchTable, err := array.TableFromJSON(memory.DefaultAllocator, wrongSchema, []string{
+		`[{"foo": true, "bar": "bar_string", "baz": "123", "qux": "2024-03-07"},
+		  {"foo": false, "bar": "bar_string", "baz": "456", "qux": "2024-03-07"}]`,
+	})
+	t.Require().NoError(err)
+	t.writeParquet(tbl.FS().(iceio.WriteFileIO), filePath, mismatchTable)
+
+	files := []string{filePath}
+
+	pterm.DisableOutput() // disable error to console
+	defer pterm.EnableOutput()
+
+	tx := tbl.NewTransaction()
+	err = tx.AddFiles(files, nil, false)
+	t.Error(err)
+	t.EqualError(err, `error encountered during parquet file conversion: error encountered during schema visitor: mismatch in fields:
+   | Table Field              | Requested Field         
+✅ | 1: foo: optional boolean | 1: foo: optional boolean
+✅ | 2: bar: optional string  | 2: bar: optional string 
+❌ | 3: baz: optional int     | 3: baz: optional string 
+✅ | 4: qux: optional date    | 4: qux: optional date   
+`)
+}
+
+func TestTableWriting(t *testing.T) {
+	suite.Run(t, &TableWritingTestSuite{formatVersion: 1})
```

Comment:
also run for V2 tables?

Reply:
bah, made it into a subsequent commit that wasn't part of this one. -_- I'll update!